### PR TITLE
Add unit tests to mutate layer fields

### PR DESCRIPTION
**Pull request by liyuqian**
So we're sure that:
1. We won't break the API by not allowing to mutate fields
2. We won't forget to markNeedsAddToScene after mutation

### CODE REVIEW

File: packages/flutter/test/rendering/layers_test.dart [hunk 1]
```diff
@@ -109,4 +111,132 @@ void main() {
     expect(leaderLayer.debugSubtreeNeedsAddToScene, true);
     expect(followerLayer.debugSubtreeNeedsAddToScene, true);
   });
+
+  void checkNeedsAddToScene(Layer layer, void mutate()) {
+    layer.debugMarkClean();
+    layer.updateSubtreeNeedsAddToScene();  // ignore: invalid_use_of_protected_member
+    expect(layer.debugSubtreeNeedsAddToScene, false);
+    mutate();
```
Comment (Hixie) [inline packages/flutter/test/rendering/layers_test.dart]: nit: rename this argument "mutateCallback" or something to make it clear that this is an argument and not a method... I just spent a few minutes searching for the "mutate()" method. :-)

File: packages/flutter/test/rendering/layers_test.dart [hunk 2]
```diff
@@ -109,4 +111,132 @@ void main() {
...
+    layer.updateSubtreeNeedsAddToScene();  // ignore: invalid_use_of_protected_member
+    expect(layer.debugSubtreeNeedsAddToScene, true);
+  }
+
+  test('mutating PictureLayer fields triggers needsAddToScene', () {
+    final PictureLayer pictureLayer = PictureLayer(Rect.zero);
+    checkNeedsAddToScene(pictureLayer, (){
```
Comment (Hixie) [inline packages/flutter/test/rendering/layers_test.dart]: nit: missing space before `{`

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/test/rendering/layers_test.dart (hunk 1 above)
Comment (liyuqian) [reply packages/flutter/test/rendering/layers_test.dart]: Done.

File: packages/flutter/test/rendering/layers_test.dart (hunk 2 above)
Comment (liyuqian) [reply packages/flutter/test/rendering/layers_test.dart]: Done.

**Pull request merged**